src/activity_handlers.cpp

```diff
@@ -1028,26 +1028,26 @@ void activity_handlers::reload_finish( player_activity *act, player *p )
         return;
     }
 
+    std::string msg = _( "You reload the %s." );
+
     if( reloadable->is_gun() ) {
         p->recoil -= act->moves_total;
 
         if( reloadable->has_flag( "RELOAD_ONE" ) ) {
             for( int i = 0; i != qty; ++i ) {
```

Comment:
> I suspect the compiler elides those and even if it doesn't it's unlikely to be a performance bottleneck. 

Most likely not. Assigning a C-String to a `std::string` will invoke dynamic memory allocation. And the `_` macro means a function from gettext will be called, which is (usually) inside a library that is loaded at runtime. The compiler can not know how that function behaves.

Anyway, the if-block should be moved out of the loop. There is nothing in it that depends on the loop variable and it will do the very same thing on each iteration.